source/crud/tests/write/findOneAndReplace-upsert.yml

```diff
@@ -3,18 +3,20 @@ data:
     - {_id: 2, x: 22}
     - {_id: 3, x: 33}
 minServerVersion: '2.6'
-# See SERVER-5289 for why the collection data is only checked for server versions >= 2.6
 
 tests:
     -
-        description: "FindOneAndReplace when no documents match with upsert returning the document before modification"
+        description: "FindOneAndReplace when no documents match without id specified with upsert returning the document before modification"
         operation:
             name: findOneAndReplace
             arguments:
                 filter: {_id: 4}
                 replacement: {x: 44}
                 projection: {x: 1, _id: 0}
-                sort: {x: 1}
+                # Omit the sort option as it has no effect when no documents
+                # match and would only cause an inconsistent return value on
+                # pre-3.0 servers when combined with returnDocument "before"
+                # (see: SERVER-17650).
```

Comment:
Per the comment, the `sort` option had no purpose here other than creating an inconsistent result value for pre-3.0 servers. It seemed prudent to simply remove it and make the test more compatible, rather than create test variants.